src/main/java/spring/intro/controllers/UserController.java

```diff
@@ -0,0 +1,54 @@
+package spring.intro.controllers;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import spring.intro.dto.UserResponseDto;
+import spring.intro.model.User;
+import spring.intro.service.UserService;
+
+@RestController
+@RequestMapping("/user")
+public class UserController {
+    private final UserService userService;
+
+    @Autowired
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @GetMapping("/inject")
+    public void injectUsers() {
+        userService.add(new User("user1", "111", "user1@mail.com"));
+        userService.add(new User("user2", "222", "user2@mail.com"));
+        userService.add(new User("user3", "333", "user3@mail.com"));
+        userService.add(new User("user4", "444", "user4@mail.com"));
+    }
+
+    @GetMapping
+    public List<UserResponseDto> getAll() {
+        return userService.listUsers().stream()
+                .map(this::mapUserToDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/{userId}")
+    public UserResponseDto get(@PathVariable Long userId) {
+        return userService.listUsers().stream()
```

Comment:
it's better to add method getById into Dao  and Service layers

Reply:
agree. done